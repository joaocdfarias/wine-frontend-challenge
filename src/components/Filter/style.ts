import styled from 'styled-components'

export const Aside = styled.aside`
  max-width: 256px;
  display: flex;
  flex-direction: column;
  gap: 32px;

  .title {
    font-size: 20px;
    font-weight: 700;
    color: #000000;
  }

  .subtitle {
    p {
      font-size: 18px;
      font-weight: 700;
      color: #333333;
      margin-bottom: 16px;
    }

    ul {
      display: flex;
      flex-direction: column;
      gap: 12px;
    }

    li {
      display: flex;
      align-items: center;
      gap: 12px;
    }

    input[type='radio'] {
      width: 16px;
      height: 16px;
    }
  }
`
