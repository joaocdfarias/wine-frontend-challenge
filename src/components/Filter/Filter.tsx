import React from 'react'
import * as S from './style'

interface IPriceFilter {
  id: number
  text: string
  min: number
  max: number
  defaultChecked?: boolean
}

const priceFilter: IPriceFilter[] = [
  {
    id: 0,
    text: 'Até R$40',
    min: 0,
    max: 40,
    defaultChecked: true,
  },
  {
    id: 1,
    text: 'R$ 40 A R$60',
    min: 40,
    max: 60,
  },
  {
    id: 2,
    text: 'R$100 A R$200',
    min: 100,
    max: 200,
  },
  {
    id: 3,
    text: 'R$200 A R$500',
    min: 200,
    max: 500,
  },
  {
    id: 4,
    text: 'Acima de R$500',
    min: 500,
    max: Infinity,
  },
]

export const Filter = () => {
  const [filterValue, setFilterValue] = React.useState({
    min: 0,
    max: 40,
  })

  const handleFilterValue = (min: number, max: number) => {
    setFilterValue({ min: min, max: max })
  }

  return (
    <S.Aside>
      <p className="title">Refine sua busca</p>
      <div className="subtitle">
        <p>Por preço</p>
        <ul>
          {priceFilter.map((price) => {
            return (
              <li key={price.id}>
                <input
                  onChange={() => handleFilterValue(price.min, price.max)}
                  type="radio"
                  name="price"
                  id={'price-' + price.id}
                  value={price.max}
                  defaultChecked={filterValue.min === price.min}
                />
                <label htmlFor={'price-' + price.id}>{price.text}</label>
              </li>
            )
          })}
        </ul>
      </div>
    </S.Aside>
  )
}
