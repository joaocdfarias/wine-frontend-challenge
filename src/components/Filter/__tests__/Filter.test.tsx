import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Filter } from '../Filter'

describe('Filter', () => {
  it('should render correctly', () => {
    const { baseElement } = render(<Filter />)

    expect(baseElement).toMatchSnapshot()
  })

  it('should select radio button correclty', async () => {
    render(<Filter />)

    const radio = screen.getByRole('radio', { name: 'R$ 40 A R$60' })
    await userEvent.click(radio)
    await waitFor(() => expect(radio).toHaveProperty('checked'))
  })
})
