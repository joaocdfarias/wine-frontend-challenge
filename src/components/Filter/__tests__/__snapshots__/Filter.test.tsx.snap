// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render correctly 1`] = `
<body>
  <div>
    <aside
      class="sc-80601c25-0 jpwTMA"
    >
      <p
        class="title"
      >
        Refine sua busca
      </p>
      <div
        class="subtitle"
      >
        <p>
          Por preço
        </p>
        <ul>
          <li>
            <input
              checked=""
              id="price-0"
              name="price"
              type="radio"
              value="40"
            />
            <label
              for="price-0"
            >
              Até R$40
            </label>
          </li>
          <li>
            <input
              id="price-1"
              name="price"
              type="radio"
              value="60"
            />
            <label
              for="price-1"
            >
              R$ 40 A R$60
            </label>
          </li>
          <li>
            <input
              id="price-2"
              name="price"
              type="radio"
              value="200"
            />
            <label
              for="price-2"
            >
              R$100 A R$200
            </label>
          </li>
          <li>
            <input
              id="price-3"
              name="price"
              type="radio"
              value="500"
            />
            <label
              for="price-3"
            >
              R$200 A R$500
            </label>
          </li>
          <li>
            <input
              id="price-4"
              name="price"
              type="radio"
              value="Infinity"
            />
            <label
              for="price-4"
            >
              Acima de R$500
            </label>
          </li>
        </ul>
      </div>
    </aside>
  </div>
</body>
`;
